pkg/strategy/production.go

```diff
@@ -125,7 +125,10 @@ func newProductionStrategy(ctx context.Context, jaeger *v1.Jaeger) S {
 
 	// prepare the deployments, which may get changed by the elasticsearch routine
 	cDep := collector.Get()
-	queryDep := inject.Sidecar(jaeger, inject.OAuthProxy(jaeger, query.Get()))
+	queryDep := inject.OAuthProxy(jaeger, query.Get())
+	if jaeger.Spec.Query.TracingEnabled == nil || *jaeger.Spec.Query.TracingEnabled == true {
+		queryDep = inject.Sidecar(jaeger, queryDep)
```

Comment:
@jpkrohling should it be necessary to explicitly add the sidecar if the annotation has been defined? If so, what is the purpose of adding the annotation?

Reply:
We can then avoid the query deployment from being created without the sidecar, only to get the sidecar injected on the next reconciliation for the `Deployment` resource.